src/vm/tests/test-snapshots.js

```diff
@@ -917,3 +980,153 @@ test('delete stopped joyent-minimal VM', function(t) {
         t.end();
     }
 });
+
+test('create joyent-minimal VM to test metadata through rollback',
+function (t) {
+    var payload = {
+        alias: 'test-snapshots-' + process.pid,
+        brand: 'joyent-minimal',
+        autoboot: false,
+        image_uuid: image_uuid,
+        do_not_inventory: true
+    };
+    vmobj = undefined;
+
+    VM.create(payload, function (err, obj) {
+        if (err) {
+            t.ok(false, 'error creating VM: ' + err.message);
+            t.end();
+        } else {
+            t.ok(true, 'VM created with uuid ' + obj.uuid);
+            VM.load(obj.uuid, function (e, o) {
+                t.ok(!err, 'loading VM after create');
+                if (!err) {
+                    t.ok(o.snapshots.length === 0, 'no snapshots after create');
+                    t.ok(o.hasOwnProperty('zfs_filesystem'),
+                        'has zfs_filesystem');
+                    vmobj = o;
+                } else {
+                    abort = true;
+                }
+                t.end();
+            });
+        }
+    });
+});
+
+test('create garbage /.zonecontrol/metadata.sock', function (t) {
+    var server;
+    var zonecontrol;
+
+    t.ok(vmobj, 'have vmobj');
+    if (!vmobj) {
+        t.end();
+        return;
+    }
+
+    zonecontrol = vmobj.zonepath + '/root/.zonecontrol/metadata.sock';
+    try {
+        fs.mkdirSync(path.dirname(zonecontrol));
```

Comment:
Good point, I'll put the unlinkSync in its own try block and change so that we're checking the exception for the ones we expect and having the test fail if there's an exception that's not EEXIST for mkdir or ENOENT for unlink. Would that resolve this?